src/Context.hs

```diff
@@ -118,6 +119,11 @@ replaceInternalEnv :: Context -> Env -> Context
 replaceInternalEnv ctx env =
   ctx {contextInternalEnv = Just env}
 
+-- The previous environment is completely replaced and will not be recoverable.
```

Comment:
This comment seems a little off since the above function also completely replaces the env?

Reply:
It turns out I just half-took the comment from the function above. Will fix.